app-libs/stf/src/trusted_call.rs

```diff
@@ -273,22 +275,44 @@ where
 				Ok(())
 			},
 
-			TrustedCall::pay_as_bid(who, orders_file) => {
+			TrustedCall::pay_as_bid(who, orders) => {
 				let now = Instant::now();
 
-				let raw_orders = fs::read_to_string(&orders_file).map_err(|e| {
-					StfError::Dispatch(format!("Error reading {}. Error: {:?}", orders_file, e))
-				})?;
-				let orders: Vec<Order> = serde_json::from_str(&raw_orders).map_err(|err| {
-					StfError::Dispatch(format!("Error serializing to JSON: {}", err))
-				})?;
+				let ORDERS_DIR = "./records/orders";
+				let RESULTS_DIR = "./records/market_results";
+
+				create_dir_all(ORDERS_DIR).unwrap();
+				create_dir_all(RESULTS_DIR).unwrap();
 
 				let market_input = MarketInput { orders: orders.clone() };
 				let orders_encoded: Vec<Vec<u8>> = orders.iter().map(|o| o.encode()).collect();
 
+				let timestamp = &orders[0].time_slot;
+
+				let orders_path = format!("{}/{}.json", ORDERS_DIR, timestamp);
+				let results_path = format!("{}/{}.json", RESULTS_DIR, timestamp);
+
+				if fs::metadata(&orders_path).is_ok() {
+					info!("Orders file already exists for timestamp {}", timestamp);
+					return Ok(())
+				}
+
 				let order_merkle_root = merkle_root::<Keccak256, _>(orders_encoded);
 				let pay_as_bid: MarketOutput = pay_as_bid_matching(&market_input);
 
+				fs::write(&orders_path, serde_json::to_string(&orders).unwrap()).map_err(|e| {
+					StfError::Dispatch(format!("Writing results {:?}. Error: {:?}", orders, e))
+				})?;
```

Comment:
I would extract helper methods for reading and writing to `best_energy_helper.rs`:


```rust
pub fn write_orders(timestamp: &str, market_results: MarketResults) -> Result<(), StfError> {
    let results_path = format!("{}/{}.json", RESULTS_DIR, timestamp);
    fs::write(&orders_path, serde_json::to_string(&orders).unwrap()).map_err(|e| {
					StfError::Dispatch(format!("Writing results {:?}. Error: {:?}", orders, e))
				})
}
```

You can do the same for results.